src/cryptonote_core/miner.h

```diff
@@ -119,8 +147,37 @@ namespace cryptonote
     bool m_do_print_hashrate;
     bool m_do_mining;
 
-  };
-}
-
+    // background mining stuffs ..
 
+    //static const int CPU_DATA_OVERALL_TIMES_COUNT = 4;
+    //static const int CPU_DATA_OVERALL_USER_INDEX = 0;
+    //static const int CPU_DATA_OVERALL_NICE_INDEX = 1;
+    //static const int CPU_DATA_OVERALL_SYSTEM_INDEX = 2;
+    //static const int CPU_DATA_OVERALL_IDLE_INDEX = 3;
+    //const int CPU_DATA_PROCESS_TIMES_COUNT;
+    //struct cpu_data
+    //{
+    //  uint64_t update_time; // set to 0 to update
+    //  uint64_t overall_cpu_times[CPU_DATA_OVERALL_TIMES_COUNT];
+    //  uint64_t process_cpu_time;
+    //};
 
+    bool background_worker_thread();
+    volatile bool m_is_background_mining_enabled;
```

Comment:
Well, drop `volatile` and make it `std::atomic`. Although I wonder - should  the setter be `public`? If it were no longer public, it could be guaranteed that the `miner` class would have to be updated for a threading issue to become a concern (and thus the `std::atomic` wrapper could be dropped). Reading from an atomic isn't "free", so if its unnecessary ...